Dockerfile

```diff
@@ -58,10 +64,37 @@ RUN jupyter labextension install nbdime-jupyterlab --no-build && \
         jupyter lab clean && \
         jlpm cache clean && \
         npm cache clean --force && \
-        rm -rf $HOME/.node-gyp && \
-        rm -rf $HOME/.local && \
-    fix-permissions $CONDA_DIR $HOME
+        rm -rf "${HOME}/.node-gyp/" && \
+        rm -rf "${HOME}/.local/" && \
+    fix-permissions "${CONDA_DIR}" "${HOME}"
+
+# READ-ONLY notebooks
+RUN mkdir -p "notebooks"
+COPY notebooks/* notebooks/
+RUN jupyter trust notebooks/*ipynb
+
+# Dont want it called work as well as ONED
+# to avoid confusion. But dont remove if its got anything in it.
+RUN rmdir work
+
+# generate non-volative disk
+RUN mkdir -p notebooks/data && fix-permissions notebooks/data
+
+###########################################
```

Comment:
This bit might need a bit of comment ("this creates a local mount point for an external folder in the host machine, useful for backups and so on")